cmd/client-s3.go

```diff
@@ -795,15 +805,47 @@ func (c *S3Client) Watch(ctx context.Context, options WatchOptions) (*WatchObjec
 	return wo, nil
 }
 
+type ttfbRecorder struct {
+	readStart time.Time
+	source    string
+	rc        io.ReadCloser
+}
+
+func (f *ttfbRecorder) Read(p []byte) (n int, err error) {
```

Comment:
@klauspost this is not for benchmarking, just for monitoring. I am not convinced as well, I don't think there are cases when TTFB will deteriorate in a MinIO cluster but it seems some users find this information useful to see.

Reply:
@vadmeste My point is that is can be wildly fluctuating and sometimes reflect zero, one or several network calls. This is highly dependent on `len(p)` and it is rather meaningless.

It isn't TTFB since we are waiting for a whole bunch of requests to complete to fill whatever size p is.

There is no easy solution for this since you have no idea of when the underlying connection has run out of a buffer.

Reply:
> My point is that is can be wildly fluctuating and sometimes reflect zero, one or several network calls

How zero ? this is only called once when mc asks to download a particular object.

Reply:
There is an incoming buffer on HTTP connections, including TCP. If your read can be satisfied from that, you never hit the network.